File: 06.Stack/stack.cpp
#include <iostream>
#define MAX_LIMIT 100
using namespace std;


class Stack{
    private:
        int top;
        int data[MAX_LIMIT];

    public:
        Stack(){
            top = 0;
        }

        void push(int item){
            if (top < MAX_LIMIT ) {
                data[top] = item;
                top = top + 1;
            } else
            {
                cout << "Stack is full!" << endl;
            }
        }

        int pop(){
            int item;
            if (top == 0) {
                cout << "Stack is emptu" << endl;
                return -1;
            } else
            {
                top = top - 1;
                item = data[top];
            }
            return item;
        }
};


int main(void){
    Stack my_stack;

    int item;

    my_stack.push(1);
    my_stack.push(2);
    my_stack.push(3);

    item = my_stack.pop();
    cout << item << endl;

    item = my_stack.pop();
    cout << item << endl;

    return 0;
}

File: 06.Stack/stack.java
public class stack {
  static final int MaxLimit = 100;
  private int[] data = new int[MaxLimit];
  private int top;

  stack(){
    this.top = 0;
  }

  public void push(int item) {
    if (this.top < stack.MaxLimit) {
      this.data[this.top] = item;
      this.top = this.top + 1;
    } else {
      System.out.println("Stack is full!");
    }
  }

  public int pop() {
    int item;

    if (this.top == 0) {
      System.out.println("Stack is empty!");
      return -1;
    } else {
      this.top = this.top - 1;
      item = this.data[this.top];
    }
    return item;
  }

  public static void main(final String[] args) {
    stack my_stack = new stack();

    my_stack.push(1);
    my_stack.push(2);
    my_stack.push(3);

    System.out.printf("%d\n", my_stack.pop());
    System.out.printf("%d\n", my_stack.pop());

  }
}